notify

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+#############################################################
+# Sample shell-script for using notify-send with matterhorn #
+#############################################################
+
+# Positional parameters
+mentioned="${1?}"
+sender="${2?}"
+message="${3?}"
+
+# Script options
+
+# notify_URGENCIES
+# First word is urgency for items where you are not mentioned, the second is
+# urgency for items where you are mentioned.  Use "none" to not be notified,
+# otherwise low, normal, critical are options
+notify_URGENCIES="none normal"
+# The desktop notification category
+notify_CATEGORY="im.received"
+# Notification header
+notify_HEAD="Matterhorn message from $sender"
+# Notification body
+notify_BODY="$message"
+
+getUrgencyHelper() {
+    shift "$mentioned"
+    printf '%s' "$1"
```

Comment:
I don't think that the original Bourne shell had `printf` support, so it's likely not available in other derivatives as well.  Your comment below indicates you are attempting to be portable, but since this is a captive script and not the user's main environment, I'd suggest it's probably fine to just explicitly use `#!/bin/bash`.  Alternatively, I think just `echo "$1"` would be safe here.

Reply:
If we're going to go with `bash`, this is more portable: `#!/usr/bin/env bash`.

Reply:
printf is quite portable, even without shell support (see e.g. `/usr/bin/printf`) but you are correct that echo is safe here; I have a habit of not using `echo "$foo"` since that breaks in various ways when `$foo` begins with a `-`